spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -443,6 +443,7 @@ private Schema lazyExpectedSchema() {
       return expectedSchema;
     }
 
+    @SuppressWarnings("checkstyle:RegexpSingleline")
```

Comment:
I kept the current implementation as `sessionState().newHadoopConf()` is more expensive. I think the correct fix here would be to broadcast the Hadoop conf. We already do that while broadcasting `FileIO`, though.

Reply:
I think we can address that in a follow-up.

Reply:
I think this is fixed in the Spark 3 path, where we use FileIO instead of creating a file system directly.

Reply:
Let's leave it as is then.